vent/sqldb/adapters/sqlite_adapter.go

```diff
@@ -0,0 +1,309 @@
+	}
+
+	i := 0
+
+	for _, tableColumn := range table.Columns {
+		cKey = 0
+		i++
+
+		// INSERT INTO TABLE (*columns).........
+		if columns != "" {
+			columns += ", "
+			insValues += ", "
+		}
+		columns += tableColumn.Name
+		insValues += "$" + fmt.Sprintf("%d", i)
+
+		if !tableColumn.Primary {
+			cKey = cols + nKeys
+			nKeys++
+
+			// INSERT........... ON CONFLICT......DO UPDATE (*updValues)
+			if updValues != "" {
+				updValues += ", "
+			}
+			updValues += tableColumn.Name + " = $" + fmt.Sprintf("%d", cKey+1)
+		}
+
+		upsertQuery.Columns[tableColumn.Name] = types.UpsertColumn{
+			IsNumeric:   tableColumn.Type.IsNumeric(),
+			InsPosition: i - 1,
+			UpdPosition: cKey,
+		}
+	}
+	upsertQuery.Length = cols + nKeys
+
+	query := fmt.Sprintf("INSERT INTO %s (%s) VALUES (%s) ", table.Name, columns, insValues)
+
+	if nKeys != 0 {
+		query += fmt.Sprintf("ON CONFLICT ON CONSTRAINT %s_pkey DO UPDATE SET ", table.Name)
+		query += updValues
+	} else {
+		query += fmt.Sprintf("ON CONFLICT ON CONSTRAINT %s_pkey DO NOTHING", table.Name)
+	}
+	query += ";"
+
+	upsertQuery.Query = query
+	return upsertQuery
+}
+
+// LastBlockIDQuery returns a query for last inserted blockId in log table
+func (adapter *SQLiteAdapter) LastBlockIDQuery() string {
+	query := `
+		WITH ll AS (
+			SELECT MAX(%s) AS %s FROM %s WHERE %s = $1 
+		)
+		SELECT COALESCE(%s, '0') AS %s 
+			FROM ll LEFT OUTER JOIN %s log ON (ll.%s = log.%s);`
+
+	return fmt.Sprintf(query,
+		types.SQLColumnNameId,          //max
+		types.SQLColumnNameId,          //as
+		types.SQLLogTableName,          //from
+		types.SQLColumnNameEventFilter, //where
+
+		types.SQLColumnNameHeight,                    //coalesce
+		types.SQLColumnNameHeight,                    //as
+		types.SQLLogTableName,                        //from
+		types.SQLColumnNameId, types.SQLColumnNameId) //on
+
+}
+
+// FindTableQuery returns a query that checks if a table exists
+func (adapter *SQLiteAdapter) FindTableQuery() string {
+	query := "SELECT COUNT(*) found FROM %s WHERE %s = $1;"
+
+	return fmt.Sprintf(query,
+		types.SQLDictionaryTableName, //from
+		types.SQLColumnNameTableName) //where
+
+}
+
+// TableDefinitionQuery returns a query with table structure
+func (adapter *SQLiteAdapter) TableDefinitionQuery() string {
+	query := `
+		SELECT 
+			%s,%s,%s,%s 
+		FROM 
+			%s 
+		WHERE 
+			%s=$1 
+		ORDER BY
+			%s;`
+
+	return fmt.Sprintf(query,
+		types.SQLColumnNameColumnName, types.SQLColumnNameColumnType, //select
+		types.SQLColumnNameColumnLength, types.SQLColumnNamePrimaryKey, //select
+		types.SQLDictionaryTableName,   //from
+		types.SQLColumnNameTableName,   //where
+		types.SQLColumnNameColumnOrder) //order by
+
+}
+
+// AlterColumnQuery returns a query for adding a new column to a table
+func (adapter *SQLiteAdapter) AlterColumnQuery(tableName string, columnName string, sqlColumnType types.SQLColumnType, length int, order int) (string, string) {
+	sqlType, _ := adapter.TypeMapping(sqlColumnType)
```

Comment:
`AlterColumnQuery(tableName string, columnName string, sqlColumnType types.SQLColumnType, length int, order int) (string, string)`
should be
`AlterColumnQuery(tableName, columnName string, sqlColumnType types.SQLColumnType, length, order int) (string, string)`
instead

Reply:
ready